src/Service/Startup.cs

```diff
@@ -58,12 +58,12 @@ public Startup(IConfiguration configuration, ILogger<Startup> logger)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            string configFileName = Configuration.GetValue<string>("ConfigFileName", RuntimeConfigLoader.DEFAULT_CONFIG_FILE_NAME);
+            string configFileName = Configuration.GetValue<string>("ConfigFileName", FileSystemRuntimeConfigLoader.DEFAULT_CONFIG_FILE_NAME);
             string? connectionString = Configuration.GetValue<string?>(
-                RuntimeConfigLoader.RUNTIME_ENV_CONNECTION_STRING.Replace(RuntimeConfigLoader.ENVIRONMENT_PREFIX, ""),
+                FileSystemRuntimeConfigLoader.RUNTIME_ENV_CONNECTION_STRING.Replace(FileSystemRuntimeConfigLoader.ENVIRONMENT_PREFIX, ""),
                 null);
             IFileSystem fileSystem = new FileSystem();
-            RuntimeConfigLoader configLoader = new(fileSystem, configFileName, connectionString);
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configFileName, connectionString);
```

Comment:
As a note, we'll need to think how this section is implemented, as at the moment it's very heavily tied to the Loader implementation, but if we want to support others, we're going to need a way to specify _which_ Loader implementation to use.

I did try and do this in the past, but there was a problem that we need to access the Provider elsewhere in the codebase (seen in line 287 for example) and we can't resolve it from the DI container as the container isn't built yet.

Reply:
Yeah this makes sense.

Reply:
will need a task for this one as well

Reply:
https://github.com/Azure/data-api-builder/issues/1593. added here.